JAM FAQ (#5804)
w3f/polkadot-wiki: docs/learn/learn-jam-faq.md

---
id: learn-jam-faq
title: FAQ for Polkadot's JAM Chain
sidebar_label: JAM FAQ
description: FAQ about Polkadot's Join-Accumulate Machine.
keywords: [Polkadot, JAM, join-accumulate, faq]
slug: ../learn-jam-faq
---

:::info Technical information about JAM

For an in-depth overview of JAM, see the [Wiki page about the JAM Chain](./learn-jam-chain.md).

:::

## What is JAM?

> JAM is a research and development project initiated by Parity Technologies. It's a computational

Reviewer: ```suggestion
JAM is a research and development project initiated by [Parity Technologies](https://www.parity.io/). It's a computational
```

model that focuses on the process of collecting, refining, joining, and accumulating data within a
blockchain network.

## What does "JAM" stand for?

"JAM" stands for "Join Accumulate Machine." It represents the core principles of the computational

> model implemented in JAM.

Reviewer: ```suggestion
model to be implemented on the JAM chain.
```

## Why is JAM groundbreaking?

The dilemma of achieving ultimate performance and scalability while still being flexible to build
any use case is as old as blockchains. Current Web3 developers have to choose whether or not to
build a smart contract or an appchain. Smart contracts on L1s are faster to write but come with
downsides. They are limited to the underlying protocol, are not as efficient, do not offer the
flexibility teams need and compete with the rest of the ecosystem for resources. Appchains provide a
more efficient and flexible development environment, as the chains are custom-made for a specific
purpose. With JAM, groundbreaking scalability currently only seen through rollups is brought to the
consensus layer. Developers no longer need to decide if they build appchains or smart contracts;

> they need a flexible and rich environment for both. So, in short, there will be L2 scalability

Reviewer: ```suggestion
with JAM, they have a flexible and rich environment for both. So, in short, there will be L2 scalability
```

without the need for rollup solutions while being fully flexible to build any application.

## What are services?

Services are modules that run on top of JAM. One of these services would be, for example, the
ChainService. This service would implement the parachain logic that is currently enshrined in the

> Polkadot protocol. JAM’s `refine` and `accumulate` phases are generic and provided by these

Reviewer: My understanding is that the generic nature of `refine` and `accumulate` phases allows for running specific services like ChainService, not the other way around.

```suggestion
Polkadot protocol. JAM’s `refine` and `accumulate` phases are generic and provide these
```

Author: yes, I rephrased

services. This will make it possible to implement even more completely permissionless services, such

> as providing an actor-like framework to run on top of JAM.

Reviewer: What is an "actor-like" framework?

Author: no idea, I deleted it

## How do services interact with JAM?

Services interact with JAM through predefined entry points: Refine, Accumulate, and onTransfer (see
more information on the [JAM Chain page](./learn-jam-chain.md#service-entry-points)). Each service
defines its specific functions and workflows, executed within the JAM framework according to the
network's requirements.

## What are the main components of JAM?

JAM consists of four main components: Refine, Accumulate, and onTransfer (see more information on
the [JAM Chain page](./learn-jam-chain.md#service-entry-points)). These components define the
workflow for processing and integrating data within the network.

## What is the goal of JAM?

JAM's primary goal is to provide a flexible and efficient framework for managing data and
computations within a network. It aims to streamline data integration and maintenance while ensuring
the network's integrity and security.

## Is JAM related to Polkadot?

Yes, JAM is a potential candidate for evolving the Polkadot Relay Chain. The key change is that
parachains are no longer enshrined; anyone can permissionlessly run their service. One service will

> be, for example, the current parachain functionality.

Reviewer: Are we implying that parachain protocol is no longer enshrined like they are currently on the relay chain and will be running on top of a service that is compatible with parachain protocol? In that case, we will have to rephrase the answer better.

> anyone can permissionlessly run their service.

Anyone can permissionlessly run which service? Won't parachain protocol be running as a system service?

## How would the Relay Chain change with JAM?

JAM would be a successor to the Relay Chain, with a more straightforward and flexible architecture.

> Only the consensus would be kept, while the rest, such as security, governance, etc., would run on

Reviewer: ```suggestion
Only the consensus functionality would be kept, while the rest, such as security, governance, etc., would run on
```

system-level services. The current Relay Chain will run on multiple cores in the new JAM
architecture, guaranteeing full compatibility.

> ## In the new architecture, what does the stack look like?

Reviewer: ```suggestion
## In the new architecture, what does the tech stack look like?
```

> From the point of view of parachains, the stack doesn’t look that much different. They will continue

Reviewer: ```suggestion
From the point of view of parachains, the tech stack doesn’t look that much different. They will continue
```

months from its presentation at Token2049 in Dubai (April 2024), not including the service layer.

## What about Polkadot? Will there be a hard fork?

JAM isn’t a successor to Polkadot but a potential evolution of the Polkadot Relay Chain. It can only
proceed if it passes through Polkadot’s decentralized governance system, winning approval from token
holders. The current parachain-centered logic would continue as a service on top of the new JAM
architecture. Part of the proposal will include tooling and hard-coded compatibility guarantees. The
Relay Chain will upgrade, and parachains can then enjoy a much richer environment, synchronous
compatibility, and excellent scaling capabilities better than current L2s but built inside the L0/L1
layer.

## Do parachains need to rewrite their code? How much work is it to build?

Parachains will be one of the “products” that can run on JAM. JAM will only remove the enshrined
parachain consensus and replace it with a model supporting different services. For the first years
after launch, it will still be faster to develop parachains, as the necessary tooling would need to

> be developed. For the migration, full compatibility guarantees will be written into the code.

Reviewer: ```suggestion
be developed. For the migration support, full compatibility guarantees will be written into the code.
```

> be developed. For the migration, full compatibility guarantees will be written into the code.

Reviewer: Clarity needed here-

For the first years
after launch, it will still be faster to develop parachains **(when compared to developing what?)**, as the necessary tooling **(for what?)** would need to
be developed.

Reviewer 2: compared to building services, as the necessary tooling for those would need to be developed.

testing, and iteration. The RFC will be published as a first step, where the community gives
feedback and iterates on it. The timeframe that JAM is expected to be production-ready is 20-60
months from its presentation at Token2049 in Dubai (April 2024), not including the service layer.

## What about Polkadot? Will there be a hard fork?

JAM isn’t a successor to Polkadot but a potential evolution of the Polkadot Relay Chain. It can only
proceed if it passes through Polkadot’s decentralized governance system, winning approval from token
holders. The current parachain-centered logic would continue as a service on top of the new JAM
architecture. Part of the proposal will include tooling and hard-coded compatibility guarantees. The
Relay Chain will upgrade, and parachains can then enjoy a much richer environment, synchronous
compatibility, and excellent scaling capabilities better than current L2s but built inside the L0/L1
layer.

## Do parachains need to rewrite their code? How much work is it to build?

Parachains will be one of the “products” that can run on JAM. JAM will only remove the enshrined

> parachain consensus and replace it with a model supporting different services. For the first years
> after launch, it will still be faster to develop parachains, as the necessary tooling would need to
> be developed. For the migration support, full compatibility guarantees will be written into the

Reviewer: ```suggestion
parachain consensus and replace it with a model that supports different services. For the first years
after launch, it will still be faster to develop parachains, as the necessary tooling for developing services would need to
be developed and adopted. For the migration support, full compatibility guarantees will be written into the
```
